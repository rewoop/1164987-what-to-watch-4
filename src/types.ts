export interface Film {
  id: number;
  filmTitle: string;
  filmVideo: string;
  filmGenre: string;
  releaseDate: number;
  backgroundPoster: string;
  filmPoster: string;
  ratingScore: number;
  ratingLevel: number;
  ratingCount: string;
  filmDescription: string;
  filmDirector: string;
  filmStarring: string[];
  filmRunTime: string;
  isFavoriteFilm: boolean;
}

export type Films = Film[];

export interface FilmComment {
  id: number;
  user: {
    id: number;
    name: string;
  };
  comment: string;
  date: string;
  rating: number;
}

export type FilmComments = FilmComment[];

export interface PostReview {
  rating: number;
  comment: string;
}
