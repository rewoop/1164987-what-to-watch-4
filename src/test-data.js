export const TestFilm = {
  id: 1,
  filmTitle: `The Rock`,
  filmGenre: `Thriller`,
  filmVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  releaseDate: 1996,
  isFavoriteFilm: true,
  runTime: `356`,
  backgroundPoster: `img/bg-the-grand-budapest-hotel.jpg`,
  filmPoster: `img/the-grand-budapest-hotel-poster.jpg`,
  ratingScore: 9,
  ratingLevel: 9,
  ratingCount: `1337 ratings`,
  filmDescription: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  filmDirector: `Wes Andreson`,
  filmStarring: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Tom Cruz`
  ],
};

export const TestFilms = [TestFilm];

export const TestComment = {
  id: 0,
  user: {
    id: 1,
    name: `Kate Muir`,
  },
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `December 24, 2016`,
  rating: 9,
};

export const TestComments = [TestComment];
