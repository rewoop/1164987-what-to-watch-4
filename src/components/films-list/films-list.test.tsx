import * as React from "react";
import * as renderer from "react-test-renderer";
import FilmsList from "./films-list";
import {Router} from "react-router-dom";
import history from "../../history";
import {Films} from "../../types";
import {TestFilms} from "../../test-data";

const filmsInfo: Films = TestFilms;

it(`Should FilmsList render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <FilmsList
            films={filmsInfo}
          />
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
