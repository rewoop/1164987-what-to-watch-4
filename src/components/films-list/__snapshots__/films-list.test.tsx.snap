// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmsList render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/film/1"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
        />
        <img
          alt="The Rock"
          height="175"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/1"
        onClick={[Function]}
      >
        The Rock
      </a>
    </h3>
  </article>
</div>
`;
